app/addons/auth/actions.js

```diff
@@ -28,88 +28,135 @@ var errorHandler = function (xhr, type, msg) {
 };
 
 
-export default {
+function login (username, password, urlBack) {
+  var promise = FauxtonAPI.session.login(username, password);
+
+  promise.then(function () {
+    FauxtonAPI.addNotification({ msg: FauxtonAPI.session.messages.loggedIn });
+    if (urlBack) {
+      return FauxtonAPI.navigate(urlBack);
+    }
+    FauxtonAPI.navigate('/');
+  });
+  promise.fail(errorHandler);
+}
 
-  login: function (username, password, urlBack) {
-    var promise = FauxtonAPI.session.login(username, password);
+function changePassword (password, passwordConfirm) {
+  var nodes = nodesStore.getNodes();
+  var promise = FauxtonAPI.session.changePassword(password, passwordConfirm, nodes[0].node);
 
-    promise.then(function () {
-      FauxtonAPI.addNotification({ msg: FauxtonAPI.session.messages.loggedIn });
-      if (urlBack) {
-        return FauxtonAPI.navigate(urlBack);
-      }
-      FauxtonAPI.navigate('/');
-    });
-    promise.fail(errorHandler);
-  },
+  promise.done(function () {
+    FauxtonAPI.addNotification({ msg: FauxtonAPI.session.messages.changePassword });
+    FauxtonAPI.dispatch({ type: ActionTypes.AUTH_CLEAR_CHANGE_PWD_FIELDS });
+  });
 
-  changePassword: function (password, passwordConfirm) {
-    var nodes = nodesStore.getNodes();
-    var promise = FauxtonAPI.session.changePassword(password, passwordConfirm, nodes[0].node);
+  promise.fail(errorHandler);
+}
 
-    promise.done(function () {
-      FauxtonAPI.addNotification({ msg: FauxtonAPI.session.messages.changePassword });
-      FauxtonAPI.dispatch({ type: ActionTypes.AUTH_CLEAR_CHANGE_PWD_FIELDS });
-    });
+function updateChangePasswordField (value) {
+  FauxtonAPI.dispatch({
+    type: ActionTypes.AUTH_UPDATE_CHANGE_PWD_FIELD,
+    value: value
+  });
+}
+
+function updateChangePasswordConfirmField (value) {
+  FauxtonAPI.dispatch({
+    type: ActionTypes.AUTH_UPDATE_CHANGE_PWD_CONFIRM_FIELD,
+    value: value
+  });
+}
 
-    promise.fail(errorHandler);
-  },
+function createAdmin (username, password, loginAfter) {
+  var nodes = nodesStore.getNodes();
+  var promise = FauxtonAPI.session.createAdmin(username, password, loginAfter, nodes[0].node);
 
-  updateChangePasswordField: function (value) {
-    FauxtonAPI.dispatch({
-      type: ActionTypes.AUTH_UPDATE_CHANGE_PWD_FIELD,
-      value: value
-    });
-  },
+  promise.then(function () {
+    FauxtonAPI.addNotification({ msg: FauxtonAPI.session.messages.adminCreated });
+    if (loginAfter) {
+      FauxtonAPI.navigate('/');
+    } else {
+      FauxtonAPI.dispatch({ type: ActionTypes.AUTH_CLEAR_CREATE_ADMIN_FIELDS });
+    }
+  });
 
-  updateChangePasswordConfirmField: function (value) {
+  promise.fail(function (xhr, type, msg) {
```

Comment:
Could you rather use the second function in the `then` promise to handle the fail case.